### PR TITLE
Fix Bug with Install demo configuration running in cluster mode with -y

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
Previous behavior prior to 2.11 would be that init security and cluster mode would only be set if they were explicitly passed in. There is a small bug that is setting them to be true if -y is passed in. This is causing a few failures with duplicate keys in integration tests when -y is passed in. 
https://github.com/opensearch-project/security/blob/2.11/tools/install_demo_configuration.sh#L73-L98

### Issues Resolved
[List any issues this PR will resolve]

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [x] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by derek-ho** [comment]
Will fix #3775 #3876

**Comment by cwperks** [comment]
Related issue: https://github.com/opensearch-project/security/issues/2402

**Comment by cwperks** [comment]
@derek-ho @DarshitChanpura What should the behavior of `-y` be for `install_demo_configuration` script? Generally, `-y` is an option in CLIs to automatically respond `yes` to all user prompts. There's a longstanding bug in the `install_demo_configuration` script with the `-y` option where it was not actually answering yes to all prompts. 

It looks like the recent switch to implementing the demo configuration installer to Java fixed the bug, but it then introduced issues in the distribution build.

**Review by cwperks — DISMISSED** [review]
Approving as this is inline with previous behavior.

**Comment by derek-ho** [comment]
> Approving as this is inline with previous behavior.

I added another test to verify this behavior.

File: src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java [hunk 1]
```diff
@@ -241,6 +241,18 @@ public void testIsStringAlreadyPresentInFile_isPresent() throws IOException {
...
+    @Test
+    public void testAssumeYesDoesNotInitializeClusterMode() throws IOException {
+        String str1 = "node.name"; // cluster_mode
+        String str2 = "plugins.security.allow_default_init_securityindex"; // init_security
+
+        installer.assumeyes = true;
+        securitySettingsConfigurer.writeSecurityConfigToOpenSearchYML();
+
+        assertThat(isKeyPresentInYMLFile(installer.OPENSEARCH_CONF_FILE, str1), is(equalTo(false)));
+        assertThat(isKeyPresentInYMLFile(installer.OPENSEARCH_CONF_FILE, str2), is(equalTo(false)));
```
Comment (cwperks) [inline src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java:253]:
```suggestion
        assertThat(isKeyPresentInYMLFile(installer.OPENSEARCH_CONF_FILE, str1), is(false));
        assertThat(isKeyPresentInYMLFile(installer.OPENSEARCH_CONF_FILE, str2), is(false));
```

File: src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java [hunk 2]
```diff
@@ -241,6 +241,18 @@ public void testIsStringAlreadyPresentInFile_isPresent() throws IOException {
         assertThat(isKeyPresentInYMLFile(installer.OPENSEARCH_CONF_FILE, str2), is(equalTo(false)));
     }
 
+    @Test
+    public void testAssumeYesDoesNotInitializeClusterMode() throws IOException {
+        String str1 = "node.name"; // cluster_mode
```
Comment (cwperks) [inline src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java:246]: nit: These variable names could be more related to their values or placed inline in the `isKeyPresentInYMLFile` call.

Comment (derek-ho) [reply src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java]: Applied suggestion in: [3b5d54e](https://github.com/opensearch-project/security/pull/3935/commits/3b5d54ea76658e84ca9214fa5212c61c7d763941)

File: src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java (hunk 1 above)
Comment (derek-ho) [reply src/test/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurerTests.java]: Done in: [60769b3](https://github.com/opensearch-project/security/pull/3935/commits/60769b3e2461377e9d0c459e519ecc9ad0a35762)

**Comment by DarshitChanpura** [comment]
> @derek-ho @DarshitChanpura What should the behavior of -y be for install_demo_configuration script? Generally, -y is an option in CLIs to automatically respond yes to all user prompts. There's a longstanding bug in the install_demo_configuration script with the -y option where it was not actually answering yes to all prompts.

This what I assumed to when writing the Java code. But turns out that the original script did nothing to `cluster_mode` and `init_security` when -y was passed which was not the case when I wrote the Java code.

> It looks like the recent switch to implementing the demo configuration installer to Java fixed the bug, but it then introduced issues in the distribution build.

Agreed that -y should inherently mark other options two options as  yes, but since it is breaking builds rn, I'm happy to revert that bug fix and address it later. After this change is merged, -y will only be reduced to choosing yes for installing demo certificates and the other two prompts will be assumed as no.


An ideal usage would be to either use -y ( which automatically assumes yes for demo certs, cluster_mode and init_security)

OR pass individual options : -d for demo certs (new option), -c for cluster mode and -i for init security modules.

where -y will always take priority. But maybe that is for another release.

**Review by cwperks — APPROVED** [review]

**Review by DarshitChanpura — APPROVED** [review]
Looks good. This existing bug which was addressed in original PR and is being reverted here, should be addressed in future.

**Pull request merged**